### ISSUE TITLE
Feature request: get images from any wiki

### ISSUE THREAD

**Issue by lucamauri** [ProfessionalWiki/WikibaseLocalMedia#4]
This extension allows getting images from the wiki where the WikiBase Repository is installed. Nevertheless I think it would be useful to be able to get images from _another_ wiki that is neither Commons, nor the local Repository. For instance, in my case, I have a MW instance acting as a Repository, and a separate MW instance with Wikibase Client: I would like to reference images in the Repo instance, but images are actually at the Client one.

Ideally, it should be possible to setup the _source_ wiki somewhere, I assume it should be enough to have the API endpoint of a remote wiki to obtain images' information.

If any of the extension programmer could point in the right direction, I would be willing to work on this modification by myself for my own purpose and eventually release the modified code.

Thanks

Luca

**Comment by kghbln** [comment]
Perhaps piggybacking into https://www.mediawiki.org/wiki/Manual:$wgForeignFileRepos which allows to connect to other wikis is a way.

**Comment by lucamauri** [comment]
> Perhaps piggybacking into https://www.mediawiki.org/wiki/Manual:$wgForeignFileRepos which allows to connect to other wikis is a way.

Thanks @kghbln , looks like a promising idea.
Can you suggest where to act on the extension's code to make it read this information and use it to get the media file?

**Comment by MTRNord** [comment]
Any progress on this? I am for this reason getting `File:Unknown data type: localMedia` I think.

**Comment by JeroenDeDauw** [comment]
That error suggests you do not have the extension enabled. Or are using a tool that does not support the new type. Either way, probably not related to this feature request.

**Comment by MTRNord** [comment]
> That error suggests you do not have the extension enabled. Or are using a tool that does not support the new type. Either way, probably not related to this feature request.

Well I cannot activate it in the specific mediawiki as the wikibase repo is running on another mediawiki instance and if I activate it on the `WikibaseClient` side of the wiki the plugin will complain about missing the repository. 

TL;DR: Client and Repo are 2 different mediawikis in this case and the plugin can only be activated on instances with the repo part of wikibase activated.

**Comment by JeroenDeDauw** [comment]
I see. Indeed currently you cannot use this extension on a wiki with ONLY Wikibase Client on it. Here is a dedicated ticket for that new feature: https://github.com/ProfessionalWiki/WikibaseLocalMedia/issues/13

**Comment by MTRNord** [comment]
> I see. Indeed currently you cannot use this extension on a wiki with ONLY Wikibase Client on it. Here is a dedicated ticket for that new feature: #13

Thanks for clarifying that it is a separate issue :)

**Comment by GreenReaper** [comment]
I agree that this would be nice. [I made a SPARQL function to generate links to an existing wiki](https://furry.wikibase.cloud/wiki/Project:SPARQL/examples#Convention_series_with_data_and_logo_URL) which I figure I could convert to an actual image if JavaScript support is enabled on Wikibase.cloud, but actually being able to reference such external images  as such would be better.

We'd probably prefer [a foreign DB repo](https://mediawiki.org/wiki/Manual:$wgForeignFileRepos#ForeignDBRepo_class), as that'd presumably be faster, once the wikibase had been integrated into the server on which the other wikis were hosted rather than Wikibase.cloud.

**Comment by hrideshmg** [comment]
Hi there, I've made a #37 to try and fix this issue. Please let me know if this meets the requirements or if I've missed anything. Thanks!
